kong/core/reports.lua

```diff
@@ -181,6 +185,15 @@ local function add_ping_value(k, v)
 end
 
 
+local function add_immutable_value(k, v)
+  v = get_report_value(v)
+  if v then
+    _buffer[#_buffer + 1] = k .. "=" .. tostring(v)
+    _buffer_immutable_idx = _buffer_immutable_idx + 1
```

Comment:
minor: we could do this increment prior to the insertion and save the `#_buffer + 1`, but not a blocker.